implementations/java-json-schema/BowtieJsonSchema.java

```diff
@@ -115,18 +138,55 @@ private void dialect(JsonNode node) throws JsonProcessingException {
       DialectRequest.class
     );
 
-    // well, setting default dialect shouldn't be necessary but why not
-    if (SpecificationVersion.DRAFT2020_12.getId().equals(dialectRequest.dialect())) {
-        validatorFactory.withDefaultDialect(new Dialects.Draft2020Dialect());
-    } else if (SpecificationVersion.DRAFT2019_09.getId().equals(dialectRequest.dialect())) {
-        validatorFactory.withDefaultDialect(new Dialects.Draft2019Dialect());
-    } else if (SpecificationVersion.DRAFT7.getId().equals(dialectRequest.dialect())) {
-      validatorFactory.withDefaultDialect(new Dialects.Draft7Dialect());
+    try {
+      if (getSpecificationVersionFor("DRAFT2020_12").getId().equals(dialectRequest.dialect())) {
```

Comment:
For this section I have a different idea :)
Maybe collect available dialects like this:
```java
Map<String, Dialect> dialectsMap = Arrays.stream(Dialects.class.getClasses())
            .filter(Dialect.class::isAssignableFrom)
            .map(Dialect.class::cast)
            .collect(Collectors.toMap(dialect -> dialect.getSpecificationVersion().getId(), Function.identity()));
```
store this map as a field and query it later for specific dialect:
```java
dialectsMap.get(dialectRequest.dialect())
```

Reply:
This way hopefully I won't have to edit the harness when I start supporting another draft version